src/librustdoc/html/static/main.js

```diff
@@ -1362,19 +1381,23 @@
                 function mergeArrays(arrays) {
                     var ret = [];
                     var positions = [];
+                    var notDuplicates = {};
```

Comment:
What's with the split between `duplicates` above and `notDuplicates` here?

Reply:
Explanation time! 😄 

`mergeArrays` is only used when we have multiple queries (for example: "str,[]"). The second one (the one using `duplicates`) allows to remove more global duplicates and is called on every search.

Reply:
That makes sense, but why the inversion? Why is one `duplicates` and one `notDuplicates`?

Reply:
I have no explanation for this... I didn't write the two functions at the same time so I wrote both without thinking to the other.